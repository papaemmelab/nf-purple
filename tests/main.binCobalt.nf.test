nextflow_process {

    name "Test Process binCobalt"
    script "main.nf"
    process "binCobalt"

    test("Should bin Cobalt output in unmatched setting") {

        when {
            params {
                cobalt_tumor_ratio_tsv = "${projectDir}/tests/data/cobalt/TEST_TUMOR.cobalt.ratio.tsv.gz"
                cobalt_tumor_ratio_pcf = "${projectDir}/tests/data/cobalt/TEST_TUMOR.cobalt.ratio.pcf"
            }
            process {
                """
                input[0] = Channel.fromPath(params.cobalt_tumor_ratio_tsv)
                input[1] = Channel.fromPath(params.cobalt_tumor_ratio_pcf)
                """
            }
        }

        then {
            assert process.success
            assert process.trace.tasks().size() == 1
            
            // check expected files
            with(process.out) {
                assert cobalt_tumor_ratio_tsv.size() == 1
                assert cobalt_tumor_ratio_pcf.size() == 1
            }
        }
    }

}
