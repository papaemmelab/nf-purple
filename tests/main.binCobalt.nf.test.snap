{
    "Should bin Cobalt without failures": {
        "content": [
            {
                "0": [
                    "TEST_TUMOR.cobalt.ratio.tsv.gz:md5,f235d46690097a000fdf7ac31bb53108"
                ],
                "1": [
                    "TEST_TUMOR.cobalt.ratio.pcf:md5,d4b0aa911e08a551e123bef8816d1256"
                ],
                "2": [
                    "TEST_NORMAL.cobalt.ratio.pcf:md5,b488cbdc3b5d17e5ab1e5712f405bb90"
                ],
                "cobalt_normal_ratio_pcf": [
                    "TEST_NORMAL.cobalt.ratio.pcf:md5,b488cbdc3b5d17e5ab1e5712f405bb90"
                ],
                "cobalt_tumor_ratio_pcf": [
                    "TEST_TUMOR.cobalt.ratio.pcf:md5,d4b0aa911e08a551e123bef8816d1256"
                ],
                "cobalt_tumor_ratio_tsv": [
                    "TEST_TUMOR.cobalt.ratio.tsv.gz:md5,f235d46690097a000fdf7ac31bb53108"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.3"
        },
        "timestamp": "2024-12-20T19:50:22.208765876"
    }
}